src/providers/SignatureProvider.tsx

```diff
@@ -19,30 +60,68 @@ const createSignatureProviderState = () => {
 
     const [isDarkMode, setIsDarkMode] = useLocalStorageState(localStorageKeys.isDarkMode, false);
 
-    /**
-     * Partial<> makes the types inside optional,
-     * meaning that updates can have any of the elements within the InputElementOptions interface but does NOT require them all
-     * @example updateName({isBold: true}) // this will only update the isBold key of the state object
-     */
-    const updateName = (updates: Partial<InputElementOptions>) => {
-        // spread (...) operator, copies (spreads) the object into a new one let a = { Hello: "world"}, let b = { ...a } will spread a into the new object that the curlies { } are creating
-        // if we spread multiple things, the second one will be spread on top of the first one
-        // ...name spreads the original state into the new object, ...updates spreads the contents on top of that one thus overriding (updating) the values
-        const copyOfName = { ...name, ...updates };
-
-        copyOfName.style = {
-            fontWeight: copyOfName.isBold ? "bold" : "normal",
-            fontStyle: copyOfName.isItalics ? "italic" : "normal",
-        };
-
-        setName(copyOfName);
+    const updateState = (updates: Partial<InputElementOptions>) => {
+        if (updates.type === "name") {
+            const copyOfName = { ...name, ...updates };
+            copyOfName.style = {
+                fontWeight: copyOfName.isBold ? "bold" : "normal",
+                fontStyle: copyOfName.isItalics ? "italic" : "normal",
+            };
+            setName(copyOfName);
+        }
+
+        if (updates.type === "email") {
+            const copyOfEmail = { ...email, ...updates };
+            copyOfEmail.style = {
+                fontWeight: copyOfEmail.isBold ? "bold" : "normal",
+                fontStyle: copyOfEmail.isItalics ? "italic" : "normal",
+            };
+            setEmail(copyOfEmail);
+        }
+        if (updates.type === "company") {
+            const copyOfCompany = { ...company, ...updates };
+            copyOfCompany.style = {
+                fontWeight: copyOfCompany.isBold ? "bold" : "normal",
+                fontStyle: copyOfCompany.isItalics ? "italic" : "normal",
+            };
+            setCompany(copyOfCompany);
+        }
+        if (updates.type === "title") {
+            const copyOfTitle = { ...title, ...updates };
+            copyOfTitle.style = {
+                fontWeight: copyOfTitle.isBold ? "bold" : "normal",
+                fontStyle: copyOfTitle.isItalics ? "italic" : "normal",
+            };
+            setTitle(copyOfTitle);
+        }
+        if (updates.type === "linkedin") {
+            const copyOfLinkedin = { ...linkedin, ...updates };
+            copyOfLinkedin.style = {
+                fontWeight: copyOfLinkedin.isBold ? "bold" : "normal",
+                fontStyle: copyOfLinkedin.isItalics ? "italic" : "normal",
+            };
```

Comment:
Same here, a ton of duplicate code  - lets create a helper function that takes in `isItalics` and `isBold` as parameters and returns the style object for us

```suggestion
            copyOfLinkedin.style = createStyle(copyOfLinkedin.isBold, copyOfLinkedin.isItalics);
```